crates/hotshot/task-impls/src/quorum_proposal/mod.rs

```diff
@@ -446,7 +446,11 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, V: Versions>
         event_sender: Sender<Arc<HotShotEvent<TYPES>>>,
     ) -> Result<()> {
         let epoch_number = self.cur_epoch;
-        let epoch_transition_indicator = if self.consensus.read().await.is_high_qc_for_last_block()
+        let epoch_transition_indicator = if self
+            .consensus
+            .read()
+            .await
+            .is_high_qc_for_epoch_transition()
```

Comment:
This returns true if the QC is for the last block.  In this case we are not in transition when it comes to proposing.  For example with epoch_height == 300 if our high QC is for block 600 then we are proposing block 601 and we are definitively must be in epoch 3 (High QC is the EQC).  

I see the Restart tests are failing/flaky and I suspect it related to this fact + the changess below to not pass in the hard coded `NotInTransition`